src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -660,4 +758,15 @@ private function hasProductWithMainImage(): bool
 
         return $images[0]['type'] === 'main' && $images[0]['path'];
     }
+
+    private function getAttributeByName($name): array
+    {
+        foreach ($this->sharedStorage->get('product_attributes') as $attribute) {
+            if ($attribute['name'] === $name) {
+                return $attribute;
+            }
+        }
+
+        throw new InvalidArgumentException('Expected a value other than null.');
+    }
```

Comment:
I would like to point out some things about this function.

1. No $name type defined
2. We are getting attribute by name (that's what the function says), but actually returning an array instead of an object - I would recommend returning an instance of a class if possible.
3. With the changes above we could filter via array_filter method.